### PR TITLE
Fix last activity

### PR DESCRIPTION
**Pull request by mvaivre**
Fix last activity date and allow to manually set a page number to access to older items more easily.

### CODE REVIEW

File: src/components/PageSwitch.tsx [hunk 1]
```diff
@@ -36,13 +40,36 @@ const PageSwitch = ({
...
+  useOnClickOutside({ ref: manualPageInputRef, handler: () => setIsSettingPageManually(false) })
+
   const handlePageSwitch = (direction: 'previous' | 'next') => {
     setPageNumber(direction === 'previous' ? currentPage - 1 : currentPage + 1)
   }
 
+  const handlePageInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.value === '') setManuallySetPage('')
+
+    const pageNumber = parseInt(e.target.value)
```
Comment (nop33) [inline src/components/PageSwitch.tsx:59]: Hm, what if the result is `NaN` or some really huge number input that cannot be parsed?

File: src/pages/AddressInfoPage/index.tsx [hunk 2]
```diff
@@ -104,8 +105,11 @@ const TransactionInfoPage = () => {
       setTxLoading(true)
       setAddressTransactions(undefined)
       try {
-        const transactionsData = await fetchAddressTransactions(client, id, pageNumber)
-        setAddressTransactions(transactionsData)
+        const firstPageTransactionData = await fetchAddressTransactions(client, id, 1)
+        const currentPageTransactionData = await fetchAddressTransactions(client, id, pageNumber)
```
Comment (nop33) [inline src/pages/AddressInfoPage/index.tsx:109]: If `pageNumber` is `1`, we are basically executing the same query twice, right?

**Review by nop33 — COMMENTED** [review]
Nice job! 👏 

I found an input that breaks the pagination :D

`10000000000000002000`

http://127.0.0.1:5173/?p=10000000000000002000

image

**Comment by mvaivre** [comment]
Let's say that it's a... nice easter egg? 🫠

**Review by nop33 — APPROVED** [review]

**Pull request merged**